syntax = "proto3";


message Empty {

}

message TutorialIndex {
    int32 value = 1;
}

message Slide {
    bytes image = 1;
    bytes bin = 2;
    bytes gltf = 3;
}

message Tutorial {
    string name = 1;
    string _id = 2;
    int32 length = 3;
    repeated Slide slides = 4;
}

message TutorialInfo {
    string name = 1;
    string _id = 2;
    int32 length = 3;
    string img_data_str = 4;
}

message TutorialsList {
    int32 length = 1;
    repeated TutorialInfo tutorials = 2;
}

service Helper{
    rpc GetTutorialsList(Empty) returns (TutorialsList) {}
    rpc GetTutorial(TutorialIndex) returns (Tutorial) {}
}
